tests/t273_agent_basic.py

```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/env python
+
+import subprocess as sp
+from time import sleep
+
+from runtest import TestBase
+
+EXEARG = 't-agent'
+
+class TestCase(TestBase):
+    def __init__(self):
+        TestBase.__init__(self, 'agent', '')
+
+    def prerun(self, timeout):
+        self.subcmd = 'record'
+        self.option = '--keep-pid -g'
+        self.exearg = EXEARG
+        record_cmd  = self.runcmd()
+        self.pr_debug("prerun command: " + record_cmd)
+        # ignore stderr as uftrace fails because data is removed when libmcount
+        # terminates (runtest waits only for it)
+        self.record_p = sp.Popen(record_cmd.split(), stdin=sp.PIPE, stderr=sp.DEVNULL)
+
+        return TestBase.TEST_SUCCESS
+
+    def setup(self):
+        sleep(.01)              # time for the agent to start
+        self.subcmd = 'live'
+        self.option = '-p %d' % self.record_p.pid
+        self.exearg = ''
+
+    def postrun(self, result):
+        self.record_p.communicate(b"^D") # target waits for a char to end
+        self.record_p.wait()
+        if result == TestBase.TEST_DIFF_RESULT: # the expected result is empty
+            result = TestBase.TEST_SUCCESS
```

Comment:
This is wierd... What's the actual output?

Reply:
I only want to test the exit value of `uftrace live --pid`, as it produces no trace output. That's the only way I could think of using the test setup: starting a target with an agent in the background, and check if the client (`uftrace live --pid`) can communicate with the agent.
The test doesn't expect an output from the `live` command, so `TEST_DIFF_RESULT` is returned. But I assume the test is successful as long as the exit code from `live` is 0.

Reply:
I guess it can get the usual output with the agent and the expected output is something like below.  Then, why not just set the output and check the return code?

```
# DURATION     TID     FUNCTION
            [266409] | main() {
   2.530 us [266409] |   getchar();
   3.037 us [266409] | } /* main */
```

Reply:
Right, I wasn't sure how to use the test structure to start multiple processes. Now I check the client return code in the `prerun()` and use the `replay` command in the `run()` so I get a trace output.